File: core/src/main/java/com/safframework/rxcache/cachestrategy/impl/CacheOnlyStrategy.java
package com.safframework.rxcache.cachestrategy.impl;

import com.safframework.rxcache.RxCache;
import com.safframework.rxcache.cachestrategy.FlowableStrategy;
import com.safframework.rxcache.cachestrategy.MaybeStrategy;
import com.safframework.rxcache.cachestrategy.ObservableStrategy;
import com.safframework.rxcache.domain.Record;
import io.reactivex.Flowable;
import io.reactivex.Maybe;
import io.reactivex.Observable;
import org.reactivestreams.Publisher;

import java.lang.reflect.Type;

/**
 * Created by tony on 2018/10/2.
 */
public class CacheOnlyStrategy implements ObservableStrategy,
        FlowableStrategy,
        MaybeStrategy {

    @Override
    public <T> Publisher<Record<T>> execute(RxCache rxCache, String key, Flowable<T> source, Type type) {

        return rxCache.<T>load2Flowable(key, type);
    }

    @Override
    public <T> Maybe<Record<T>> execute(RxCache rxCache, String key, Maybe<T> source, Type type) {

        return rxCache.<T>load2Maybe(key, type);
    }

    @Override
    public <T> Observable<Record<T>> execute(RxCache rxCache, String key, Observable<T> source, Type type) {

        return rxCache.<T>load2Observable(key, type);
    }
}

File: rxjava/rxjava2/src/main/kotlin/com/safframework/rxcache/rxjava/rxjava2/transformstrategy/impl/CacheOnlyStrategy.kt
package com.safframework.rxcache.rxjava.rxjava2.transformstrategy.impl

import com.safframework.rxcache.RxCache
import com.safframework.rxcache.domain.Record
import com.safframework.rxcache.rxjava.rxjava2.load2Flowable
import com.safframework.rxcache.rxjava.rxjava2.load2Maybe
import com.safframework.rxcache.rxjava.rxjava2.load2Observable
import com.safframework.rxcache.rxjava.rxjava2.transformstrategy.FlowableStrategy
import com.safframework.rxcache.rxjava.rxjava2.transformstrategy.MaybeStrategy
import com.safframework.rxcache.rxjava.rxjava2.transformstrategy.ObservableStrategy
import io.reactivex.BackpressureStrategy
import io.reactivex.Flowable
import io.reactivex.Maybe
import io.reactivex.Observable
import org.reactivestreams.Publisher
import java.lang.reflect.Type

/**
 *
 * @FileName:
 *          com.safframework.rxcache.rxjava.rxjava2.transformstrategy.impl.CacheOnlyStrategy
 * @author: Tony Shen
 * @date: 2021-02-19 13:48
 * @version: V1.0 只获取缓存的策略
 */
class CacheOnlyStrategy : ObservableStrategy, FlowableStrategy, MaybeStrategy {

    override fun <T> execute(rxCache: RxCache, key: String, source: Flowable<T>, type: Type): Publisher<Record<T>> {
        return rxCache.load2Flowable(key, type)
    }

    override fun <T> execute(
        rxCache: RxCache,
        key: String,
        source: Flowable<T>,
        type: Type,
        backpressureStrategy: BackpressureStrategy
    ): Publisher<Record<T>> {
        return rxCache.load2Flowable(key, type, backpressureStrategy)
    }

    override fun <T> execute(rxCache: RxCache, key: String, source: Maybe<T>, type: Type): Maybe<Record<T>> {
        return rxCache.load2Maybe(key, type)
    }

    override fun <T> execute(
        rxCache: RxCache,
        key: String,
        source: Observable<T>,
        type: Type
    ): Observable<Record<T>> {
        return rxCache.load2Observable(key, type)
    }
}